From fbda7d9155dae1776994798dac849858b561955c Mon Sep 17 00:00:00 2001
From: Andrew Morgan <andrew@amorgan.xyz>
Date: Wed, 4 Dec 2024 18:43:32 +0000
Subject: [PATCH] Use a different variable when changing var type

just mypy being particular about changing the type of an already-defined variable.

Resolves the following mypy errors:

synapse/http/replicationagent.py:92: error: Incompatible types in assignment (expression has type "_WrapperEndpoint", variable has type "HostnameEndpoint")  [assignment]
synapse/http/proxyagent.py:394: error: Incompatible types in assignment (expression has type "_WrapperEndpoint", variable has type "HostnameEndpoint")  [assignment]
---
 synapse/http/proxyagent.py       | 3 ++-
 synapse/http/replicationagent.py | 4 +++-
 2 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/synapse/http/proxyagent.py b/synapse/http/proxyagent.py
index 36cb9285df..c91cf30fd1 100644
--- a/synapse/http/proxyagent.py
+++ b/synapse/http/proxyagent.py
@@ -391,7 +391,8 @@ def http_proxy_endpoint(
     if scheme == b"https":
         if tls_options_factory:
             tls_options = tls_options_factory.creatorForNetloc(host, port)
-            proxy_endpoint = wrapClientTLS(tls_options, proxy_endpoint)
+            wrapped_proxy_endpoint = wrapClientTLS(tls_options, proxy_endpoint)
+            return wrapped_proxy_endpoint, credentials
         else:
             raise RuntimeError(
                 f"No TLS options for a https connection via proxy {proxy!s}"
diff --git a/synapse/http/replicationagent.py b/synapse/http/replicationagent.py
index ee8c707062..4eabbc8af9 100644
--- a/synapse/http/replicationagent.py
+++ b/synapse/http/replicationagent.py
@@ -89,7 +89,7 @@ class ReplicationEndpointFactory:
                 location_config.port,
             )
             if scheme == "https":
-                endpoint = wrapClientTLS(
+                wrapped_endpoint = wrapClientTLS(
                     # The 'port' argument below isn't actually used by the function
                     self.context_factory.creatorForNetloc(
                         location_config.host.encode("utf-8"),
@@ -97,6 +97,8 @@ class ReplicationEndpointFactory:
                     ),
                     endpoint,
                 )
+                return wrapped_endpoint
+
             return endpoint
         elif isinstance(location_config, InstanceUnixLocationConfig):
             return UNIXClientEndpoint(self.reactor, location_config.path)
-- 
2.46.2

